packages/mobile/src/pincode/authentication.test.ts

```diff
@@ -0,0 +1,86 @@
+import * as Keychain from 'react-native-keychain'
+import { expectSaga } from 'redux-saga-test-plan'
+import { select } from 'redux-saga/effects'
+import { PincodeType } from 'src/account/reducer'
+import { pincodeTypeSelector } from 'src/account/selectors'
+import { navigate, navigateBack } from 'src/navigator/NavigationService'
+import { getPasswordSaga, getPincode } from 'src/pincode/authentication'
+import { mockAccount } from 'test/values'
+
+const mockPepper = { password: '0000000000000000000000000000000000000000000000000000000000000001' }
+const mockPin = '111555'
+
+describe(getPasswordSaga, () => {
+  const mockedNavigate = navigate as jest.Mock
+
+  it('Gets password', async () => {
+    mockedNavigate.mockImplementationOnce((_, params) => {
+      expect(params.withVerification).toBe(true)
+      params.onSuccess(mockPin)
+    })
+
+    const mockGetGenericPassword = Keychain.getGenericPassword as jest.Mock
+    mockGetGenericPassword.mockResolvedValue(mockPepper)
+    const expectedPassword = mockPepper.password + mockPin
+
+    await expectSaga(getPasswordSaga, mockAccount, true, false)
+      .provide([[select(pincodeTypeSelector), PincodeType.CustomPin]])
+      .returns(expectedPassword)
+      .run()
+
+    expect(navigate).toHaveBeenCalled()
+    expect(navigateBack).toHaveBeenCalled()
+  })
+})
+
+describe(getPincode, () => {
+  // it('returns PIN from phone keystore', async () => {
```

Comment:
Copied over from the old `getPincode` tests in the now deleted account/saga.test file